vispy/plot/_plotwidget.py

```diff
@@ -0,0 +1,254 @@
+            vertices, faces = read_mesh(fname)[:2]
+        if meshdata is not None:
+            if not all(x is None for x in (vertices, faces, fname)):
+                raise ValueError('vertices, faces, and fname must be None if '
+                                 'fname is not None')
+        else:
+            meshdata = MeshData(vertices, faces)
+        mesh = Mesh(meshdata=meshdata, vertex_colors=vertex_colors,
+                    face_colors=face_colors, color=color, shading='smooth')
+        self.add(mesh)
+        self._set_camera(TurntableCamera, azimuth=0, elevation=0)                
+        return mesh
+
+    @quick
+    def plot(self, data, **kwargs):
+        """Plot a series of data using lines and markers
+
+        Parameters
+        ----------
+        data : array | two arrays
+            Arguments can be passed as ``(Y,)``, ``(X, Y)`` or
+            ``np.array((X, Y))``.
+        color : instance of Color
+            Color of the line.
+        symbol : str
+            Marker symbol to use.
+        line_kind : str
+            Kind of line to draw. For now, only solid lines (``'-'``)
+            are supported.
+        width : float
+            Line width.
+        marker_size : float
+            Marker size. If `size == 0` markers will not be shown.
+        edge_color : instance of Color
+            Color of the marker edge.
+        face_color : instance of Color
+            Color of the marker face.
+        edge_width : float
+            Edge width of the marker.
+
+        Returns
+        -------
+        line : instance of LinePlot
+            The line plot.
+
+        See also
+        --------
+        marker_types, LinePlot
+        """
+        line = LinePlot(data, connect='strip', **kwargs)
+        self.add(line)
+        self._set_camera(PanZoomCamera)
+        return line
+
+    @quick
+    def spectrogram(self, x, n_fft=256, step=None, fs=1., cmap='cubehelix',
+                    clim='auto'):
+        """Calculate and show a spectrogram
+
+        Parameters
+        ----------
+        x : array-like
+            1D signal to operate on. ``If len(x) < n_fft``, x will be
+            zero-padded to length ``n_fft``.
+        n_fft : int
+            Number of FFT points. Much faster for powers of two.
+        step : int | None
+            Step size between calculations. If None, ``n_fft // 2``
+            will be used.
+        fs : float
+            The sample rate of the data.
+        cmap : str
+            Colormap name.
+        clim : str | tuple
+            Colormap limits. Should be ``'auto'`` or a two-element tuple of
+            min and max values.
+
+        Returns
+        -------
+        spec : instance of Image
+            The spectrogram.
+
+        See also
+        --------
+        Image
+        """
+        # XXX once we have axes, we should use the "fft_freqs", too
+        data = stft(x, n_fft, step, fs)
```

Comment:
That would also be nice because you could eventually make it so the step and such could be adjusted, which would be nice. I'll turn it into a visual like I did for Histogram earlier.